pkg/cmd/util/cmd.go

```diff
@@ -31,3 +33,29 @@ func GetDisplayFilename(filename string) string {
 
 	return filename
 }
+
+// ResolveResource returns the resource name and kind of the resourceString.
+// If the resource string has no specified kind, defaultKind will be returned.
+func ResolveResource(defaultKind, resourceString string, mapper meta.RESTMapper) (string, string, error) {
+	if mapper == nil {
+		return "", "", errors.New("mapper cannot be nil")
+	}
+
+	var name string
+	parts := strings.Split(resourceString, "/")
+	switch len(parts) {
+	case 1:
+		name = parts[0]
+	case 2:
+		_, kind, err := mapper.VersionAndKindForResource(parts[0])
+		if err != nil {
+			return "", "", err
+		}
+		name = parts[1]
+		return kind, name, nil
```

Comment:
once we know the resource is good, and have the canonical Kind, I *think* we want to convert it back to resource form to return it using `KindToResource`